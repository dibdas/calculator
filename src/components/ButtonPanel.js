/* eslint-disable react/prop-types */
import React from 'react';
import Button from './Button';
import propTypes from 'prop-types';

const ButtonPanel = ({ clickHandler }) => {
  const handleClick = (buttonName) => clickHandler(buttonName);
  return (
    <div className="buttonPanel">
      <div className="buttonsrow">
        <Button clickHandler={handleClick} name="AC" />
        <Button name="+/-" />
        <Button name="%" />
        <Button name="รท" />
      </div>
      <div className="buttonsrow">
        <Button name="7" />
        <Button name="8" />
        <Button name="9" />
        <Button name="X" />
      </div>
      <div className="buttonsrow">
        <Button name="-" />
        <Button name="4" />
        <Button name="5" />
        <Button name="6" />
      </div>
      <div className="buttonsrow">
        <Button name="1" />
        <Button name="2" />
        <Button name="3" />
        <Button name="+" />
      </div>
      <div className="buttonsrow">
        <Button name="0" />
        <Button name="." />
        <Button name="=" />
      </div>
    </div>
  );
};

ButtonPanel.propTypes = {
  clickHandler: propTypes.func.isRequired,
};
export default ButtonPanel;
