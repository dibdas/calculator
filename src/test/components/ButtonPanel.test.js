import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPanel from '../../components/ButtonPanel';

describe('ButtonPanel', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ButtonPanel name="2" clickHandler={test} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
