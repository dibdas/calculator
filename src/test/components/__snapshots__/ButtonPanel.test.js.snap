// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel renders correctly 1`] = `
<div
  className="buttonPanel"
>
  <div
    className="buttonsrow"
  >
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
  </div>
  <div
    className="buttonsrow"
  >
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
  </div>
  <div
    className="buttonsrow"
  >
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
    </div>
  </div>
  <div
    className="buttonsrow"
  >
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
  </div>
  <div
    className="buttonsrow"
  >
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        0
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
